connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -222,7 +234,18 @@ private SchemaBuilder convertFieldType(Schema.Type type) {
             default:
                 throw new DataException("Unexpected type in Cast transformation: " + type);
         }
+    }
 
+    private static Object encodeLogicalType(Schema schema, Object value) {
```

Comment:
Err.. to clarify, we are talking about the `fromLogical`, `toLogical` methods on the Date, Decimal, Time, and Timestamp classes? I can see value in making these non-static, part of a LogicalType interface. Or, are we talking about the new encodeLogicalType(), and existing castTo* methods? These I feel are more implementation details of Cast and should remain private if possible.

Reply:
No, I'm talking more about the `encode*` and `castTo*` methods in this class. I'm not suggesting we make them public, but instead I'm asking whether it would make sense to make these *protected* and *non-static* so that it would be easier to create subclasses.

I guess the problem with that, though, is that they are implementation details and not part fo the public API. If anybody did create a subclass and we later changed the implementation, their subclass would no longer compile and they'd have problems running on different versions of Connect.

Given that, I think it's fine the way it is: private non-static or private static doesn't really matter except for subclasses.